src/nearby-stop-places/NearbyStopPlacesScreenComponent.tsx

```diff
@@ -82,16 +84,9 @@ export const NearbyStopPlacesScreenComponent = ({
         location?.resultType === 'favorite') &&
       location?.layer === 'venue'
     ) {
-      onSelectStopPlace(location);
+      onSelectStopPlaceRef.current(location);
     }
-    /*
-     Refreshing (by dragging down) doesn't work since the location id doesn't
-     change. Take a look at fixing this when looking at the exchaustive-deps
-     error. May also consider using react-query instead of the custom reducer
-     with side effect flow.
-    */
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [location?.id]);
+  }, [location]);
```

Comment:
I'm a little scared of this change 😬 Might be ok, but remember to test it thoroughly with saved location, search results location and my position location.

Reply:
no longer relevant after refactoring